module-2/backend/week_one.md

```diff
@@ -1,34 +1,64 @@
+5. What types of variables are accessible in our view templates without explicitly passing them?
+Instance variables
+6. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
+
   ```ruby
   get '/horses' do
-    erb :index
+    @count = 1
+    name = "Mr. Ed"
+    erb :index :locals => name: name
   end
   ```
 
-8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+7. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
 9. What's the purpose of ERB?
+The purpose of Embedded Ruby is to insert ruby into a text document, usually into an HTML page.
 10. Why do I need a development AND test database?
+You need to have two databases so that you can separate your your development and testing suite. For testing you would probably want to use fixtures to cut down on the time that your tests take. In development you might want to use a fuller database to closely mirror what your application will be using.
 11. What's responsive design?
+Responsive design is opposed to fixed design. In my mind, a fixed design would be the same across all different platforms and clients. Whereas a responsive design changes dynamically to fit the needs of each individual client/platform. An example would be a screen that would resize itself across smartphones, desktops etc.
 12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
+CRUD stands for Create, Read, Update, and Delete. This is important because these describes the way in which a user can interact with a given database.
+13. What does HTTP stand for?
+  Hypertext Transfer Protocol
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
 15. What's an ORM?
+Object Relational Mapper, used to create uniform databases from mismatched datasets.
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+Active Record, because it's awesome.
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
+  get '/restaurant'
+  get '/restaurant/:id'
+  get '/restaurant/new'
+  post '/restaurant'
+  get '/restaurant/:id/edit'
+  put '/restaurant/:id'
+  Delete '/restaurant/:id'
+18. What's a migration?
+It is a way of altering the structure of your database.
 19. When you create a migration, does it automatically modify your database?
+ No it does not automatically modify your database, but rather your schema.

```

Comment:
does not modify schema until the command is run to run the actual migrations (rake db:migrate)